metadata-models/src/main/pegasus/com/linkedin/ml/metadata/TrainingData.pdl

```diff
@@ -0,0 +1,24 @@
+namespace com.linkedin.ml.metadata
+
+import com.linkedin.common.DatasetUrn
+
+/**
+ * Ideally, the MLModel card would contain as much information about the training data as the evaluation data. However, there might be cases where it is not feasible to provide this level of detailed information about the training data. For example, the data may be proprietary, or require a non-disclosure agreement. In these cases, we advocate for basic details about the distributions over groups in the data, as well as any other details that could inform stakeholders on the kinds of biases the model may have encoded.
+ */
+record TrainingData {
```

Comment:
Maybe we should also capture the job (perhaps `AzkabanJobUrn`) that is performing the training execution? If we do add the job urn, then input datasets need not be added in `TrainingData` aspect, they will be captured in the `AzkabanJobAspect`. Thoughts? cc @hshahoss

Reply:
1. We don't want to tie in with `Azkaban` as the only supported scheduler
2. With the help of `DataprocessSnapshot` this info is available, we could always link it using the graph relationship to the MLModel Training Data.

Reply:
Agree. `DataProcess` entity might be a better candidate than `AzkabanJob`. But `DataProcess` itself has `DataProcessInfo` aspect that includes list of dataset urns as input. The list of urns we have in `TrainingData` should ideally be the same as provided in `DataProcessInfo`? Then mentioning it here might not be useful?

We can keep the list of dataset urns here for now. But would love to hear your thoughts on the redundancy aspect of this, particularly once we include `DataProcessUrn`.

Reply:
In my understanding `DataProcess` and `MLModels` will be part of separate `MCE` messages even sent by separate processes. In that case the only way to tie these together is using the `DatasetUrn`. 
Also, having the link through `DatasetUrn` might be better than directly through the `DataProcess` as the MLEngineers who work with the `MlModels` know the underlying dataset very well than the process that created it. Thoughts?

Reply:
So the `DataProcess` I am referring to is the process/job that is training the ML model. ML engineers should be aware of the same? Again, we can add it later since we might not have support for `DataProcess` that is responsible for training ML models.
The same was referenced in this paper http://learningsys.org/nips17/assets/papers/paper_13.pdf as "Training Run" entity. Let me know your thoughts.

Reply:
Resolving this, we can always add a new field to capture training execution later.